tests/unit/test_database_memory.py

```
@@ -65,6 +65,475 @@
+        
+        # First connection
+        conn1 = reloaded_database._connect()
+        assert conn1 is not None
+        assert reloaded_database._MEMORY_CONNECTION is not None
+        
+        # Second connection should be the same object
+        conn2 = reloaded_database._connect()
+        assert conn2 is conn1
+        assert conn2 is reloaded_database._MEMORY_CONNECTION
+
+    def test_connect_with_uri_memory_database(self, monkeypatch, restore_database_module):
```

Ruff (0.14.4)

Unused method argument: `restore_database_module`

(ARG002)